### PR TITLE
EVG-2506 streamline test history query a bit

### PR DESCRIPTION
**Pull request by john-m-liu**

### CODE REVIEW

File: model/task_history.go [hunk 1]
```diff
@@ -776,68 +776,48 @@ func testHistoryV2Results(params *TestHistoryParameters) ([]task.Task, error) {
...
+	wg := sync.WaitGroup{}
+	wg.Add(numQueryThreads)
+	resultChan := make(chan task.Task)
+	for i := 0; i < numQueryThreads; i++ {
+		go func() {
+			defer wg.Done()
+			for t := range taskChan {
+				err := t.MergeNewTestResults() //nolint
+				if err != nil {
+					grip.Error(errors.Wrapf(err, "error merging test results for task %s", t.Id))
```
Comment (tychoish) [inline model/task_history.go]: do we want to catcher these?

File: model/task_history.go [hunk 2]
```diff
@@ -776,68 +776,48 @@ func testHistoryV2Results(params *TestHistoryParameters) ([]task.Task, error) {
...
+		for t := range resultChan {
+			out = append(out, t)
 		}
-	}
-
-	return tasks, nil
+	}()
+	wg.Wait()
+	close(resultChan)
+	<-doneChan
```
Comment (tychoish) [inline model/task_history.go]: I think if you buffer the output channel, you can get rid of the second command?

File: model/task_history.go (hunk 1 above)
Comment (john-m-liu) [reply model/task_history.go]: added

File: model/task_history.go (hunk 2 above)
Comment (john-m-liu) [reply model/task_history.go]: I don't know the size of the output channel ahead of time

**Pull request merged**